java-checks/src/main/java/org/sonar/java/checks/MathClampMethodsCheck.java

```diff
@@ -0,0 +1,159 @@
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.ConditionalExpressionTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import static org.sonar.java.model.ExpressionUtils.skipParentheses;
+
+@Rule(key = "S6885")
+public class MathClampMethodsCheck extends IssuableSubscriptionVisitor implements JavaVersionAwareVisitor {
+
+  public static final String CONDITIONAL_EXPRESSION_MESSAGE = "Use \"Math.clamp\" instead of a conditional expression.";
+  public static final String METHOD_INVOCATION_MESSAGE = "Use \"Math.clamp\" instead of \"Math.min\" or \"Math.max\".";
+
+  public static final String JAVA_LANG_MATH = "java.lang.Math";
+  private static final MethodMatchers MATH_MIN_METHOD_MATCHERS = MethodMatchers.create()
+    .ofTypes(JAVA_LANG_MATH)
+    .names("min")
+    .withAnyParameters()
+    .build();
+
+  private static final MethodMatchers MATH_MAX_METHOD_MATCHERS = MethodMatchers.create()
+    .ofTypes(JAVA_LANG_MATH)
+    .names("max")
+    .withAnyParameters()
+    .build();
+
+  @Override
+  public boolean isCompatibleWithJavaVersion(JavaVersion version) {
+    return version.isJava21Compatible();
+  }
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return List.of(Tree.Kind.CONDITIONAL_EXPRESSION, Tree.Kind.METHOD_INVOCATION);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (tree.is(Tree.Kind.CONDITIONAL_EXPRESSION)) {
+      checkConditionalExpression((ConditionalExpressionTree) tree);
+    } else if (tree.is(Tree.Kind.METHOD_INVOCATION)) {
+      checkMethodInvocation((MethodInvocationTree) tree);
+    }
+  }
+
+  private void checkConditionalExpression(ConditionalExpressionTree firstConditionalExpression) {
+    BinaryExpressionTree condition = greaterOrLessBinaryExpression(firstConditionalExpression.condition());
+    if (condition != null) {
+      boolean isGreater = isGreaterThanOrEqual(condition);
+      var trueExpression = skipParentheses(firstConditionalExpression.trueExpression());
+      var falseExpression = skipParentheses(firstConditionalExpression.falseExpression());
+      if (shouldReportOnConditional(condition.rightOperand(), trueExpression, falseExpression, isGreater) ||
+        shouldReportOnConditional(condition.rightOperand(), falseExpression, trueExpression, !isGreater)) {
+        reportIssue(firstConditionalExpression, CONDITIONAL_EXPRESSION_MESSAGE);
+      }
+    }
+  }
+
+  private static boolean shouldReportOnConditional(ExpressionTree condition, ExpressionTree tree1, ExpressionTree tree2, boolean isMax) {
+    if (condition.is(Tree.Kind.IDENTIFIER) && tree1.is(Tree.Kind.IDENTIFIER) && (((IdentifierTree) condition).name().equals(((IdentifierTree) tree1).name()))) {
```

Comment:
Comparing `String IdentifierTree.name()` should not create false-positives here, but it's not always the case and could confuse the reader. The safe way to test is with `!identifierTree1.symbol().isUnknown() && identifierTree1.symbol().equals(identifierTree2.symbol())`.
I didn't find a matching implementation in the helpers, only the opposite: `SyntacticEquivalence.areDifferentVariables`.
It could be useful to add one helper, like `ExpressionUtils.areSameVariables(Tree tree1, Tree tree2, boolean defaultValue)`. In this case, we could even support `Tree.Kind.MEMBER_SELECT`. It would be possible to check `this.field` is same as `field`.

Same comment for the bellow `checkInnerExpression` method.